include/extractor/guidance/intersection_handler.hpp

```diff
@@ -508,15 +508,15 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
             // even reverse the direction. Since we don't want to compute actual turns but simply
             // try to find whether there is a turn going to the opposite direction of our obvious
             // turn, this should be alright.
-            NodeID new_node;
-            const auto previous_intersection = [&]() {
-                EdgeID turn_edge;
-                std::tie(new_node, turn_edge) = intersection_generator.SkipDegreeTwoNodes(
+            const auto previous_intersection = [&]() -> IntersectionView {
+                const auto parameters = intersection_generator.SkipDegreeTwoNodes(
```

Comment:
Ah I see -- the graph walker does too much here and we don't want to pay the incurred costs. How would a specialized graph walker look like: simply not generating intermediate `Intersection`s for skipping? Could this decision be a policy for the graph walker? And are there other use-cases for not generating intersections while walking the graph? (If not maybe just leave it as is)